src/main/java/com/Elessar/database/MongoDB.java

```diff
@@ -0,0 +1,111 @@
+package com.Elessar.database;
+
+import com.Elessar.app.server.User;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import java.util.ArrayList;
+import java.util.List;
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.eq;
+
+/**
+ * Created by Hans on 1/20/19.
+ */
+public class MongoDB implements MyDatabase{
+    private final MongoDatabase db;
+    public MongoDB(MongoDatabase db) {
+        this.db = db;
+    }
+
+    @Override
+    public void insert(User user) {
+        final Document doc = new Document();
+        doc.append("name", user.getName());
+        doc.append("password", user.getPassword());
+        doc.append("email", user.getEmail());
+        doc.append("online", user.getOnline());
+        if (user.getPhoneNumber().length() > 0) {
```

Comment:
It seems protobuf will use a empty string to represent missing optional filed

Reply:
Sure protobuf use empty strings.  But you don't know if the User objects are always created out of protobuf. Here all we have is a User object, it doesn't matter where it comes from and how it's created.
If you want to avoid a check here, you have to add an assertion inside User class when phone number get assigned

Reply:
make sense